webchild_sampling.py

```diff
@@ -0,0 +1,92 @@
+from urllib.request import urlopen, urlretrieve
+from io import BytesIO
+from zipfile import ZipFile
+from mowgli.lib.etl.web_child.web_child_transformer import WebChildTransformer
+from mowgli.lib.cskg.edge import Edge
+from mowgli.lib.cskg.node import Node
+from mowgli.lib.cskg.mowgli_predicates import WN_SYNSET
+from random import sample
+import csv
+from mowgli.lib.etl.pipeline_wrapper import PipelineWrapper
+from mowgli.lib.etl.web_child.web_child_pipeline import WebChildPipeline
+from mowgli.lib.etl.pipeline_storage import PipelineStorage
+
+
+
+def loadcsv():
+
+    print("Downloading Zip")
+    url = urlopen('http://people.mpi-inf.mpg.de/~ntandon/resources/relations/partOf/webchild_partof.zip')
+    urlstream = BytesIO(url.read())
+
+    z = ZipFile(urlstream)
+    print("Extracting Zip")
+    z.extractall()
+    
+    print("Downloading Glossary")
+    wordnet = urlretrieve('http://people.mpi-inf.mpg.de/~ntandon/resources/relations/metadata/noun.gloss','noun.gloss')
+
+    transformer = WebChildTransformer()
+
+    print('transforming')
+    triplegen = transformer.transform(memberof_csv_file_path ="webchild_partof_memberof.txt",
+        physical_csv_file_path="webchild_partof_physical.txt",
+        substanceof_csv_file_path="webchild_partof_substanceof.txt",
+        wordnet_csv_file_path="noun.gloss")
+
+
+    nodemap = dict()
+    edgemap = dict()
+    part_whole_edges = list()
+    nodes = dict()
+
+    print('mapping edges and nodes')
+    for node_or_edge in triplegen:    
+        
+        if isinstance(node_or_edge, Node):
+            node = node_or_edge
+            nodes[node.id] = node
+        elif isinstance(node_or_edge, Edge):
+            edge = node_or_edge
+            if edge.predicate == WN_SYNSET:
+                continue
+            else:
+                part_whole_edges.append(edge)
+        else:
+            raise TypeError
+
+    selected_edges = sample(part_whole_edges, k= 200)
```

Comment:
The samples in the CSV are all related to anatomy, which seems improbable given the size of the WebChild data set.